### ISSUE TITLE
./gravity-sync.sh compare fails with "Validating Connection to 192.168.178.144timeout: unrecognized option: preserve-status" after fresh install

### ISSUE THREAD

**Issue by coza73** [vmstan/gravity-sync#135]
**Issue Description**

Brand new install and when I run ./gravity-sync.sh compare it fails with the following error.

pihole2:~/gravity-sync# ./gravity-sync.sh compare
[∞] Gravity Sync 3.2.4 Executing
[✓] Importing gravity-sync.conf Settings
[✓] Evaluating Arguments: COMPARE Requested
[i] Targeting root@192.168.178.144
[✓] Validating SSH Client on pihole2 (OpenSSH)
[✓] Validating RSYNC Installed on pihole2
[✓] Validating Gravity Sync Folders on pihole2
[✓] Validating Pi-hole Configuration
[e] Validating Connection to 192.168.178.144timeout: unrecognized option: preserve-status
BusyBox v1.31.1 () multi-call binary.

Usage: timeout [-s SIG] SECS PROG ARGS

Runs PROG. Sends SIG to it if it is not gone in SECS seconds.
Default SIG: TERM.
[✗] Validating Connection to 192.168.178.144

**Configuration Details**

Pihole  running as docker container in Alpine Linux alpine-standard-3.12.1-x86_64

pihole2:~/gravity-sync# ./gravity-sync.sh info
[∞] Gravity Sync 3.2.4 Executing
[✓] Importing gravity-sync.conf Settings
[✓] Evaluating Arguments: INFO Requested
========================================================
Local Software Versions
Gravity Sync 3.2.4
Pi-hole
  Pi-hole version is v5.2.4 (Latest: v5.2.4)
  AdminLTE version is v5.3.1 (Latest: v5.3.1)
  FTL version is v5.5.1 (Latest: v5.5.1)
Linux 5.4.84-0-virt x86_64
bash 5.0.17(1)-release
OpenSSH_8.3p1, OpenSSL 1.1.1i  8 Dec 2020
rsync  version 3.1.3  protocol version 31
sqlite3 3.32.1 2020-05-25 16:19:56 0c1fcf4711a2e66c813aed38cf41cd3e2123ee8eb6db98118086764c4ba83350
Sudo version 1.9.0
git version 2.26.2
Docker version 19.03.12, build 48a66213fe1747e8873f849862ff3fb981899fc6

Local/Secondary Instance Settings
Local Hostname: pihole2
Local Pi-hole Type: docker
Local Pi-hole Config Directory: /data
Local DNSMASQ Config Directory: /data/dnsmasq.d
Local Pi-hole Container Name: pihole
Local Docker Binary Directory: /usr/bin/docker
Local File Owner Settings: 999:999
DNS Replication: Disabled (custom)
CNAME Replication: Disabled (default)
Verify Operations: Disabled (custom)
Ping Test: Enabled (default)
Backup Retention: 7 days (default)

Remote/Primary Instance Settings
Remote Hostname/IP: 192.168.178.144
Remote Username: root
Remote Pi-hole Type: docker
Remote Pi-hole Config Directory: /data
Remote DNSMASQ Config Directory: /data/dnsmasq.d
Remote Pi-hole Container Name: pihole
Remote Docker Binary Directory: /usr/bin/docker
Remote File Owner Settings: 999:999
========================================================
[∞] Gravity Sync INFO Aborting (1 seconds)

**Comment by coza73** [comment]
If I remove --preserve-status from gs-ssh.sh line 13 it seems to work just fine

**Comment by vmstan** [comment]
Interesting, I wonder if the --preserve-status option is missing from the Alpine implementation.

What is out output of `timeout --version` on that system?

**Comment by coza73** [comment]
Yeah I gathered as much.      Is not having --preserve status going to hurt me in this case?

The alpine build of timeout does not even have --version implemented.

Anyway I have worked out it is running coreutils-8.32-r0.      

Only reason I am running alpine at this point is as an extremely light docker host.     I might be moving over to photon-os soon so that my fix this current issue

**Comment by nh-mike** [comment]
Hi @coza73 
Alpine uses an older version of Busybox, which misses the --preserve-status option.

You can either update to a newer version of busybox, or you can install the coreutils package which includes a version of timeout which has --preserve-status implemented.

I built a docker container upon alpine and had this very same problem and tested both of these options and both worked for me. You may also use the Busybox container as a base - I ripped the busybox binary over into my alpine container and it worked however in the end I just installed coreutils.

**Comment by coza73** [comment]
@nh-mike you are correct I assumed coreutils was installed since timeout command was already there.    Once I actually installed coreutils the 'correct' version of timout was installed.

**Comment by vmstan** [comment]
> Yeah I gathered as much. Is not having --preserve status going to hurt me in this case?

It's mostly used when issuing commands via SSH to the remote Pi-hole. The original reason I put it in there, was I didn't want to lose the resulting status of the remote command. Since the whole script basically runs sequentially, each step has to complete successfully for the next operation to either be able to function or know what to do. My thought at the time was if you issue a remote command, and the command itself fails, but timeout says "hey I sent it, that's all you're expecting" and returns a completed error code, then the next part of the script could possibly fail.

It's interesting that it works without it, it's entirely possible it was never doing what I intended for it to do.

> I might be moving over to photon-os soon so that my fix this current issue.

I have been running my production Gravity Sync with a Docker container of Pi-hole, on Photon OS, for a while. Can confirm it works great. (I also work for VMware, so I'm a little biased.)

I did recently move it to a Ubuntu box to be able to better test some other things.
